PEP 817: Final set of pre-split updates (#4867)
python/peps: peps/pep-0817.rst

non-variant wheels as well.

> Suggested implementation logic per type of packaging tool

Reviewer: Can we simplify this language?

```suggestion
Suggested implementation logic for a packaging tool
```

---------------------------------------------------------

Installing a package from an index
''''''''''''''''''''''''''''''''''

    .. figure:: pep-0817/conceptual_diagram_installers.svg
        :target: _images/conceptual_diagram_installers.svg

>         :alt: TODO.

Reviewer 2: Please add alt text.

Reviewer 3: Thanks for the quick response @hugovk. We'll update.

>         A conceptual diagram of installing a wheel.

Reviewer 2: Unindent to remove blockquote:

<img width="766" height="747" alt="Image" src="https://github.com/user-attachments/assets/17f95767-c31a-4021-8351-b9851c697498" />

```suggestion
.. figure:: pep-0817/conceptual_diagram_installers.svg
    :target: _images/conceptual_diagram_installers.svg
    :alt: TODO.

    A conceptual diagram of installing a wheel.
```

When asked to install a version of a package from an index, the proposed

> behavior would be to:

Reviewer: ```suggestion
When asked to install a version of a package from an index, the proposed
tool behavior would be to:
```

> 1. Query the remote index for the package in question.

Reviewer: ```suggestion
1. Query the remote index for the desired package.
```

2. Initially select a package version meeting the version constraints,

>    as usual (this does not need to take variant metadata into account).

Reviewer: ```suggestion
2. Select an initial match for a package version meeting the version constraints,
   as usual (this does not need to take variant metadata into account).
```

3. Filter available wheels based on Platform Compatibility Tags.

> 4. Determine if any of the remaining wheels are variant wheels.

Reviewer: ```suggestion
4. Determine if any of the filtered wheels are variant wheels.
```

Author: I'm not a native speaker but this sounds ambiguous to me. It could be misinterpreted as "filtered out".

>    If not, proceed as with non-variant wheels.

Reviewer: > proceed as with non-variant wheels

This would use the initial match. Correct?

Author: The "initial match" above is for a version. You still need to select among the wheels for that version.

     - otherwise, if the provider is vendored or reimplemented, query it
       in implementation-specific manner.
     - otherwise, if the Python provider package is considered secure
       (either by the installer or via explicit user opt-in), install it
       in an isolated environment, and query it via the plugin API.
     - if none of the above applies, do not run the provider and either
       consider the variant properties incompatible, or fail the
       installation.

   - for the disabled providers (e.g. opt-in providers that were not
     enabled by the user, providers excluded via environment markers),
     assume that all variant properties in the namespace are
     incompatible.

8. Filter and order variants based on the lists of supported properties,
   and select the most preferred variant. If no variant wheel matched,
   use the non-variant wheels by their rules.
9. If multiple wheels for a given version share the same variant label,
   order them by Platform compatibility tags and build number, and
   select the best wheel.

Installing a local wheel
''''''''''''''''''''''''

When asked to install a local wheel file, the proposed behavior would be

> to:

Reviewer: ```suggestion
When asked to install a local wheel file, the tool's proposed behavior would be
to:
```
